pkg/controllers/dnspolicy/dnsrecord_eventhandler.go

```diff
@@ -0,0 +1,80 @@
+package dnspolicy
+
+import (
+	"context"
+
+	"github.com/Kuadrant/multi-cluster-traffic-controller/pkg/_internal/slice"
+	"github.com/Kuadrant/multi-cluster-traffic-controller/pkg/apis/v1alpha1"
+	"github.com/Kuadrant/multi-cluster-traffic-controller/pkg/controllers/gateway"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type DNSRecordEventHandler struct {
+	client      client.Client
+	hostService gateway.HostService
+}
+
+var _ handler.EventHandler = &DNSRecordEventHandler{}
+
+// Create implements handler.EventHandler
+func (eh *DNSRecordEventHandler) Create(e event.CreateEvent, q workqueue.RateLimitingInterface) {
+	eh.enqueueForObject(e.Object, q)
+}
+
+// Delete implements handler.EventHandler
+func (eh *DNSRecordEventHandler) Delete(e event.DeleteEvent, q workqueue.RateLimitingInterface) {
+	eh.enqueueForObject(e.Object, q)
+}
+
+// Generic implements handler.EventHandler
+func (eh *DNSRecordEventHandler) Generic(e event.GenericEvent, q workqueue.RateLimitingInterface) {
+	eh.enqueueForObject(e.Object, q)
+}
+
+// Update implements handler.EventHandler
+func (eh *DNSRecordEventHandler) Update(e event.UpdateEvent, q workqueue.RateLimitingInterface) {
+	eh.enqueueForObject(e.ObjectNew, q)
+}
+
+func (eh *DNSRecordEventHandler) enqueueForObject(obj v1.Object, q workqueue.RateLimitingInterface) {
+	ctx := context.Background()
+
+	dnsRecord, ok := obj.(*v1alpha1.DNSRecord)
+	if !ok {
+		return
+	}
+
+	dnsPolicyList := &v1alpha1.DNSPolicyList{}
```

Comment:
This looks like it will list all policies in the cluster?  Should list only policies in the same namespace as the DNSRecord I believe

Reply:
Should we restrict them to be only in the same namespace? At the moment I'm creating the DNSPolicy in the tenant namespace, whereas the DNSRecord is created in the system namespace, so that wouldn't work

Reply:
I think with the change to leverage OCM we should start with the expectation that MZ and DNSRecord will exist in the same namespace. So lets change this as part of a follow up. At some stage if we have the usecase we may look to introduce a MZ being shared / bound  to other namespces. For example one potential option would be a `subdomain` resource that references and zone resource. Similar to how a HTTPRoute references a Gateway cc @mikenairn do this make sense to you also?  
It may also be worth considering whether MZ should be a cluster level resource.